src/pcre2_compile_class.c

```diff
@@ -2292,15 +2303,13 @@ while (*ptr == META_ECLASS_NOT)
   negated = !negated;
   }
 
+*pptr = ptr;
```

Comment:
Moving up the pptr assignment breaks the pattern with the other functions. Although, it looks like it behaves the same.

Reply:
It is an optimization. There is no need to cache the pptr.